app/src/main/java/io/appium/uiautomator2/utils/ScreenshotHelper.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.uiautomator2.utils;
+
+import android.app.UiAutomation;
+import android.graphics.Bitmap;
+import android.graphics.Rect;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+
+import java.io.ByteArrayOutputStream;
+
+import io.appium.uiautomator2.common.exceptions.CompressScreenshotException;
+import io.appium.uiautomator2.common.exceptions.CropScreenshotException;
+import io.appium.uiautomator2.common.exceptions.ElementNotVisibleException;
+import io.appium.uiautomator2.common.exceptions.TakeScreenshotException;
+import io.appium.uiautomator2.model.internal.CustomUiDevice;
+
+public class ScreenshotHelper {
+
+    /**
+     * Grab device screenshot and crop it to specifyed area if cropArea is not null.
+     * Compress it to PGN format and convert to Base64 byte-string.
+     *
+     * @param cropArea  Area to crop.
+     * @return  Base64-encoded screenshot string.
+     */
+    public static String takeScreenshot(@Nullable final Rect cropArea) throws TakeScreenshotException, CompressScreenshotException, ElementNotVisibleException, CropScreenshotException {
+        Bitmap screenshot = takeDeviceScreenshot();
+        try {
+            if (cropArea != null) {
+                final Bitmap elementScreenshot = crop(screenshot, cropArea);
+                screenshot.recycle();
```

Comment:
this class is redundant

Reply:
I didn't get you. Can you explain more details?

Reply:
the `recycle` method will be called twice on `screenshot` instance if `cropArea != null` according to the current implemntation

Reply:
I can't see any chance to get double recycle call here, because we reassign `screenshot` variable after `crop` and `recycle`.

Reply:
ah, I see